src/main.rs

```diff
@@ -937,21 +938,69 @@ fn do_cmd_certify(
                 });
         }
         CertifyKind::Delta { from, to } => {
-            let importable = from.git_rev.is_none() && to.git_rev.is_none();
+            let from_is_git_version = from.git_rev.is_some();
+            let kind = AuditKind::Delta { from, to };
+            let importable = kind.default_importable();
+
+            let mut entry = AuditEntry {
+                kind,
+                criteria,
+                who,
+                importable,
+                notes,
+                aggregated_from: vec![],
+                is_fresh_import: false,
+            };
+
+            // Collapse a delta audit with a git `from` version with a prior audit that is
+            // non-importable and has identical and satisfied criteria.
+            //
+            // We merge an adjacent audit for a prior version with the new audit (updating the new
+            // audit). The later `update_store` call will remove the now-unused prior audit.
+            if from_is_git_version && !sub_args.no_collapse {
+                // A closure which returns whether the given audit entry satisfies the criteria
+                // being certified.
+                let is_rooted_for_criteria = {
+                    let mapper = CriteriaMapper::new(&store.audits.criteria);
+                    let criteria = mapper.criteria_from_list(&entry.criteria);
+                    let audit_graph = resolver::AuditGraph::build(store, &mapper, &package, None)
+                        .map_err(|_| CertifyError::BadAuditGraph)?;
+
+                    move |audit: &AuditEntry| {
+                        let version = match &audit.kind {
+                            AuditKind::Delta { from, .. } => from,
+                            AuditKind::Full { version } => version,
```

Comment:
For a full audit, the effective "from" edge is the `None` version, which satisfies all criteria, so you could kinda return `true` here so long as you check that the audit's criteria matches first. Right now you're checking the `to` version instead, which is a bit weird.

Reply:
True. In theory checking the `from` or the `to` doesn't matter much, it's just introducing one extra thing to the search path. But we could just return `true`.